lib/config/deployment-config.ts

```diff
@@ -0,0 +1,113 @@
+import { getSecret } from "../utils/sm";
+
+interface InjectedConfigOptions {
+  project: string;
+  stage: string;
+  region?: string;
+}
+
+type InjectedConfigProperties = {
+  bootstrapUsersPasswordArn: string;
+  oktaMetadataUrl: string;
+  brokerString: string;
+  iamPath: string;
+  iamPermissionsBoundary: string;
+  vpcName: string;
+};
+
+export type DeploymentConfigProperties = InjectedConfigProperties & {
+  isDev: boolean;
+  project: string;
+  sharedOpenSearchDomainArn: string;
+  sharedOpenSearchDomainEndpoint: string;
+  stage: string;
+  terminationProtection: boolean;
+};
+
+export class DeploymentConfig {
+  public config: DeploymentConfigProperties;
+
+  private constructor(config: DeploymentConfigProperties) {
+    this.config = config;
+  }
+
+  public static async fetch(
+    options: InjectedConfigOptions
+  ): Promise<DeploymentConfig> {
+    const injectedConfig = await DeploymentConfig.loadConfig(options);
+    const appConfig: DeploymentConfigProperties = {
+      ...injectedConfig,
+      project: options.project,
+      stage: options.stage,
+      isDev: !["main", "val", "production"].includes(options.stage),
```

Comment:
seds currently uses master/val/production. If we're swapping to main this lines up